src/Exception/UnsupportedValueException.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace MongoDB\Exception;
+
+use InvalidArgumentException;
+
+use function get_debug_type;
+use function sprintf;
+
+class UnsupportedValueException extends InvalidArgumentException implements Exception
```

Comment:
I noticed this wasn't in the Codec namespace. SGTM as this seems generally useful for casts where the argument isn't invalid simply because of its type and the actual value might be relevant.

For the existing cases in InvalidArgumentException, I think we only care about the type itself so there's no need to attach the value to the exception.

Reply:
Ack - I figured a new namespace for this one exception might be overkill.